### PR TITLE
Create 'export method' component. #391.

### LINKED ISSUE #391: Create "export method" component
Screen Shot 2022-08-03 at 9 53 14 PM

#### Approach
* Add an `ExportMethod` component to `/components/export`.
* Component takes the following props:
  * title - string
  * description - string (let's just allow a single text-only paragraph for now, we can polish this later)
  * route - string, optional
  * buttonLabel - string

#### Definition of Done
* Component matches [mocks](https://www.figma.com/file/i4uZsYkKAML9ivi2TIo6ug/explore-dashboard?node-id=1012%3A15549).
* If `link` is not specified, show button in disabled state.

### PR DESCRIPTION
**Pull request by frano-m**
### Ticket
Closes #391 .

### Reviewers
@MillenniumFalconMechanic .

### Changes
- Added export method component and corresponding SB.
  - export method `props` are updated as per discussion (added `disabled` and made `route` compulsory). 
- Modified some Section styles as part 1 of refactoring section component and corresponding styles.

### CODE REVIEW

File: explorer/app/components/Project/components/Section/components/SectionTitle/sectionTitle.tsx [hunk 1]
```diff
@@ -3,15 +3,16 @@ import { Typography } from "@mui/material";
 import React from "react";
 
 interface Props {
+  className?: string;
   title: string;
 }
 
-export const SectionTitle = ({ title }: Props): JSX.Element => {
+export const SectionTitle = ({ className, title }: Props): JSX.Element => {
```
Comment (frano-m) [inline explorer/app/components/Project/components/Section/components/SectionTitle/sectionTitle.tsx:10]: Updated SectionTitle to take additional styles (margin bottom) for non collapsable sections.

File: explorer/app/components/Project/components/Section/section.styles.ts [hunk 2]
```diff
@@ -1,8 +1,18 @@
+import { SectionTitle as Title } from "./components/SectionTitle/sectionTitle";
 import styled from "@emotion/styled";
-import { ButtonBase, Typography } from "@mui/material";
 import { BREAKPOINT } from "../../../../hooks/useBreakpointHelper";
+import { ButtonBase, Typography } from "@mui/material";
 
 export const Section = styled.div`
```
Comment (frano-m) [inline explorer/app/components/Project/components/Section/section.styles.ts:6]: Added basic non collapsable section. Will update names of collapsable section styles with the refactor of section.

File: explorer/app/components/Project/components/Section/components/SectionTitle/sectionTitle.tsx [hunk 3]
```diff
@@ -3,15 +3,16 @@ import { Typography } from "@mui/material";
...
   return (
     <Typography
+      className={className}
       color="ink"
       component="h3"
-      fontWeight={600}
```
Comment (frano-m) [inline explorer/app/components/Project/components/Section/components/SectionTitle/sectionTitle.tsx:14]: Font weight will be resolved with [ticket #287](https://github.com/clevercanary/data-browser/issues/287)

File: explorer/app/components/Project/components/Section/section.styles.ts (hunk 2 above)
Comment (MillenniumFalconMechanic) [reply explorer/app/components/Project/components/Section/section.styles.ts:6]: Apologies for not getting a ticket to you for a TODO here. I have created a stub ticket [here](https://github.com/clevercanary/data-browser/issues/400); can you add a corresponding TODO?

File: explorer/app/components/Project/components/Section/components/SectionTitle/sectionTitle.tsx (hunk 1 above)
Comment (MillenniumFalconMechanic) [reply explorer/app/components/Project/components/Section/components/SectionTitle/sectionTitle.tsx:10]: Thanks for the detail here, F! Is there a way we can get around suppling a class name as a prop? For example, can we push the knowledge of the margin bottom to where the collapsable functionality resides?

File: explorer/app/components/Project/components/Section/section.styles.ts (hunk 2 above)
Comment (frano-m) [reply explorer/app/components/Project/components/Section/section.styles.ts:7]: Done

File: explorer/app/components/Project/components/Section/components/SectionTitle/sectionTitle.tsx (hunk 1 above)
Comment (frano-m) [reply explorer/app/components/Project/components/Section/components/SectionTitle/sectionTitle.tsx:10]: The classname props is added for the benefit of styling a component... see https://emotion.sh/docs/styled#styling-any-component

Comment (MillenniumFalconMechanic) [reply explorer/app/components/Project/components/Section/components/SectionTitle/sectionTitle.tsx:10]: Thanks Fran!

**Review by MillenniumFalconMechanic — APPROVED** [review]
LGTM, thanks Fran!

**Pull request merged**